### PR TITLE
Simulate: Application State Change and Hash of Executed Bytecode

### PR DESCRIPTION
**Pull request by ahangsu**
Adds support and tests for the following new simulate features:
* algorand/go-algorand#5658
* algorand/go-algorand#5659
* algorand/go-algorand#5686

See https://github.com/algorand/algorand-sdk-testing/pull/303 for step definitions

REST models updated by rerunning https://github.com/algorand/generator (specifically `scripts/generate_typescript.sh`)

### CODE REVIEW

File: .test-env
Comment (ahangsu) [inline .test-env:3]:
```suggestion
SDK_TESTING_BRANCH="V2"
```

**Review by algorandskiy — COMMENTED** [review]
LGTM, please update the branch after sdk-testing merge

File: src/client/v2/algod/models/types.ts
Comment (jasonpaulos) [reply src/client/v2/algod/models/types.ts:791]: Added a link to the generator repo in the PR description: https://github.com/algorand/generator

**Review by algorandskiy — APPROVED** [review]

**Pull request merged**
